### ISSUE TITLE
keybase email forwarding not working ("User unknown in virtual alias table")

### ISSUE THREAD

**Issue by larsks** [keybase/keybase-issues#941]
I tried sending an email to my keybase address (larsks@keybase.io), and received:

```
   ----- The following addresses had permanent fatal errors -----
<larsks@keybase.io>
    (reason: 550 5.1.1 <larsks@keybase.io>: Recipient address rejected: User unknown in virtual alias tab
+le)

   ----- Transcript of session follows -----
... while talking to mail1.keybase.io.:
>>> DATA
<<< 550 5.1.1 <larsks@keybase.io>: Recipient address rejected: User unknown in virtual alias table
550 5.1.1 <larsks@keybase.io>... User unknown
<<< 554 5.5.1 Error: no valid recipients
```

Is something broken?  Or did I grossly misread something?  I thought I read the email forwarding was supposed to work now.

Thanks!

**Comment by hiergiltdiestfu** [comment]
Any update on this? Still does not work, but the CLI urges the user to add the keybase uid to the key. What's the holdup? :)

**Comment by maxtaco** [comment]
@hiergiltdiestfu it should work for you, but only as of when you verified your email address (which was yesterday).  let me know if not...

**Comment by hiergiltdiestfu** [comment]
Yes, I can confirm that it works now. I faintly remember getting an email verification request yesterday, which irritated me because I'm pretty pretty sure that I got (and follow up on) one right after I signed up. I did not keep either of these emails, tho, so I'm not sure 100% one way or the other. 

The take-away is that this line should be deleted from the success-popup on the Web-Interface version of Encrypt:
> Coming soon: $n@keybase.io will work, from this page or your mail client.

And maybe put a lil whatsitcalled on a users own profile view reflecting the verification status of that account's primary email address.

**Comment by larsks** [comment]
I wonder if an error message along the lines of "This user has not verified their email address" would be more helpful than `Recipient address rejected: User unknown in virtual alias table`.

**Comment by blalor** [comment]
My email address *is* verified, and I get the same error.

**Comment by maxtaco** [comment]
Hi Folks, we disabled this feature.  It was too much of a security and administrative hassle.  Now we recommend that people embed their real email addresses in their PGP key.

**Comment by melix** [comment]
Correct me if I'm wrong, but this is problematic when you want to use your keybase GPG key to sign commits on GitHub. I tried to do this, I added my key on GitHub, but then it wants to verify my keybase.io email. Since I never receive the notification, GitHub doesn't validate the key, so the commits are not trusted.

**Comment by melix** [comment]
To be even clearer, here is the problem:

I have a signed commit, and I have used my Keybase GPG key:

![unverified-1](https://cloud.githubusercontent.com/assets/316357/14346522/e64b26ac-fcb2-11e5-947f-1592d0263cf0.png)

You can see that it says unverified, but I have uploaded my key on GitHub. This key has multiple email addresses as identity:

![unverified-2](https://cloud.githubusercontent.com/assets/316357/14346551/0e58601a-fcb3-11e5-9068-c1f75967ccda.png)

All of those emails are verified, except the keybase one. We can't verify it, because it's no real email, and doesn't forward to my primary email:

![unverified-3](https://cloud.githubusercontent.com/assets/316357/14346594/3a858604-fcb3-11e5-81ed-2ecd6497d65e.png)

Any idea how to workaround this?

**Comment by melix** [comment]
Update: when I first signed my commit, I didn't have uploaded my GPG key yet. Now, I have done the following in the meantime:

- associate more emails to my GPG id
- re-upload the key (those 2 steps were done when the screenshot was taken)
- **push a new signed commit**

And now it shows "verified". So it seems to be a GitHub synchronization issue!

**Comment by zQueal** [comment]
Apparently you could have also told Git which subkey to use;

    git config --global user.signingkey {sub-key-id}

If you were to use another validated email address, beside your `@keybase.io` identity, then it also would have come up "Verified" without any issues.

**Comment by ahmadnassri** [comment]
as @zQueal mentioned, for the github usecase you can simply add your email address used with github to keybase key, import that key into github, and simply setup git to sign using the email address:

```
git config --global user.signingkey name@domain.tld
```
